src/mercury_engine_data_structures/common_types.py

```diff
@@ -127,6 +130,7 @@ def _emitparse(self, code):
 
         block = f"""
             def {fname}(io, this):
+                # {self.name} ({self})
```

Comment:
what is this for?

Reply:
oh, just makes it slightly easier to debug the generated files

Reply:
ah ok